operators/src/source/ogr_source.rs

```diff
@@ -207,24 +187,7 @@ pub enum OgrSourceErrorSpec {
 
 #[derive(Clone, Debug)]
 pub struct OgrSourceState {
-    dataset_information: Arc<OgrSourceDataset>,
-    default_geometry: OgrSourceDefaultGeometry,
-}
-
-#[derive(Clone, Debug)]
-pub struct OgrSourceDefaultGeometry(Option<TypedGeometry>);
-
-impl OgrSourceDefaultGeometry {
-    pub fn try_into_geometry<G>(&self) -> Result<Option<G>>
-    where
-        G: Geometry,
-    {
-        if let Some(v) = self.0.clone() {
-            Ok(Some(G::try_from(v)?))
-        } else {
-            Ok(None)
-        }
-    }
+    dataset_information: Box<dyn LoadingInfo<OgrSourceDataset, VectorResultDescriptor>>,
```

Comment:
> jdroenner commented on 20.01.2021
can'T we use a generic here and throw a box in it if we need one?

Reply:
the execution context gives us a `Box<dyn LoadingInfo<...>>`. How shall I convert this to a concrete type? I tried making `OgrState` generic over D where `D: LoadingInfo<OgrSourceDataset, VectorResultDescriptor>` but when I put the box into it it says 
> doesn't have a size known at compile-time

Reply:
you could try +?Sized. I did something like this in the playground: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=5e6550d77a93ce386ea2745502e41adf

Reply:
I would say if we can't optimize (generify) it for our core engine, we could leave it as it is then.

Reply:
If i understand you correctly this would again just hide the fact that the T is boxed, so I vote against it.